File: 202. Happy Number/Solution.c
bool isHappy(int n) {
    int x = n, y = n;
    while ( 1 ) {
        x = cal(x);
        y = cal(cal(y));
        if ( 1 == x ) return 1;
        if ( 1 == y ) return 1;
        if ( x == y ) return 0;
    }
}

int cal(int n) {
    int sum = 0;
    while ( n != 0 ) {
        sum += (n % 10) * (n % 10);
        n /= 10;
    }
    return sum;
}

File: 202. Happy Number/Solution.py
class Solution(object):
    def isHappy(self, n):
        """
        :type n: int
        :rtype: bool
        """
        hash_map = {}
        while True:
            n = sum([int(v)**2 for v in str(n)])
            if n == 1:
                return True
            if n not in hash_map:
                hash_map[n] = 1
            else:
                return False

